src/go/rdctl/cmd/factoryReset.go

```diff
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
 		if err != nil {
 			return err
 		}
-		// Note that this command's only flag is default to not remove k8s cache
-		// but the server takes an optional flag meaning the opposite (as per issues 1701 and 2408)
-		factoryResetData.KeepSystemImages = !removeKubernetesCache
-		jsonBuffer, err := json.Marshal(factoryResetData)
+		cmd.SilenceUsage = true
+		_, err = doShutdown()
 		if err != nil {
 			return err
 		}
-		result, err := processRequestForUtility(doRequestWithPayload("PUT", versionCommand("", "factory_reset"), bytes.NewBuffer(jsonBuffer)))
-		if err != nil {
-			return err
-		}
-		fmt.Println(string(result))
-		return nil
+		return map[string]func() error{
+			"darwin":  deleteDarwinData,
+			"linux":   deleteLinuxData,
+			"windows": deleteWindowsData,
+		}[runtime.GOOS]()
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(factoryResetCmd)
 	factoryResetCmd.Flags().BoolVar(&removeKubernetesCache, "remove-kubernetes-cache", false, "If specified, also removes the cached Kubernetes images.")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.Verbose, "verbose", false, "Be verbose")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.DontWait, "dont-wait", false, "Don't wait for shutdown to be confirmed")
```

Comment:
A boolean flag that's default-true would be better here (that is, the user would need to use `--wait=false` instead).

Among other things, I got confused parsing `dont`… made me think of 🍩

Reply:
I know what you mean, but this option is only useful when you specify an explicit `--wait=false` option. Specifying `--wait` is the same as `--wait=true`, which is the default setting. Usually boolean flags work the other way (although not always).